diff --git a/dist/hooks/useInterceptPopState.js b/dist/hooks/useInterceptPopState.js
index 9fbf9f7b0027fdde96177e20e4802e5dbc14887f..059cc00e09b73c301410005685aff1af144cfb2c 100644
--- a/dist/hooks/useInterceptPopState.js
+++ b/dist/hooks/useInterceptPopState.js
@@ -42,8 +42,8 @@ function useInterceptPopState({ guardMapRef, }) {
 function createHandlePopState(guardMapRef, writeState) {
     let dispatchedState;
     return (nextState) => {
-        const token = nextState.__next_navigation_guard_token;
-        const nextIndex = Number(nextState.__next_navigation_guard_stack_index) || 0;
+        const token = nextState && nextState.__next_navigation_guard_token;
+        const nextIndex = Number(nextState && nextState.__next_navigation_guard_stack_index) || 0;
         if (!token || token !== renderedStateRef.current.token) {
             if (debug_1.DEBUG)
                 console.log(`useInterceptPopState(): token mismatch, skip handling (current: ${renderedStateRef.current.token}, next: ${token})`);
diff --git a/dist/utils/historyAugmentation.js b/dist/utils/historyAugmentation.js
index 610d5fe7fe59aa575adad6a0a05a2bbb7c9641c6..7745ad77bc38390226bb92c36f229c86621e4be4 100644
--- a/dist/utils/historyAugmentation.js
+++ b/dist/utils/historyAugmentation.js
@@ -26,10 +26,11 @@ function setupHistoryAugmentationOnce({ renderedStateRef, }) {
             originalReplaceState,
         };
     }
+    const initialState = window.history.state || {};
     renderedStateRef.current.index =
-        parseInt(window.history.state.__next_navigation_guard_stack_index) || 0;
+        parseInt(initialState.__next_navigation_guard_stack_index) || 0;
     renderedStateRef.current.token =
-        String((_a = window.history.state.__next_navigation_guard_token) !== null && _a !== void 0 ? _a : "") ||
+        String((_a = initialState.__next_navigation_guard_token) !== null && _a !== void 0 ? _a : "") ||
             newToken();
     if (debug_1.DEBUG)
         console.log(`setupHistoryAugmentationOnce: initial currentIndex is ${renderedStateRef.current.index}, token is ${renderedStateRef.current.token}`);
@@ -43,8 +44,8 @@ function setupHistoryAugmentationOnce({ renderedStateRef, }) {
         };
         originalReplaceState.call(window.history, modifiedState, "", window.location.href);
     };
-    if (window.history.state.__next_navigation_guard_stack_index == null ||
-        window.history.state.__next_navigation_guard_token == null) {
+    if (initialState.__next_navigation_guard_stack_index == null ||
+        initialState.__next_navigation_guard_token == null) {
         writeState();
     }
     window.history.pushState = function (state, unused, url) {
